Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -1,73 +1,341 @@
 ﻿using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Runtime.Serialization.Json;
 using System.Text;
-using OfficeOpenXml;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
 
-namespace Streetcode.WebApi.Utils;
-
-public class WebParsingUtils
+namespace Streetcode.WebApi.Utils
 {
-    public static async void ParseZipFileFromWeb(object? state)
+    public class WebParsingUtils
     {
-        var projRootDirectory = Directory.GetParent(Environment.CurrentDirectory)?.FullName!;
+        public static async Task DownloadAndExtractAsync(string fileUrl, string zipPath, string extractTo, CancellationToken cancellationToken, ILogger logger = null)
+        {
+            if (string.IsNullOrEmpty(fileUrl))
+            {
+                throw new ArgumentException("FileUrl cannot be null or empty", nameof(fileUrl));
+            }
+
+            if (!Uri.IsWellFormedUriString(fileUrl, UriKind.Absolute))
+            {
+                throw new ArgumentException("Invalid FileUrl", nameof(fileUrl));
+            }
+
+            if (string.IsNullOrEmpty(zipPath))
+            {
+                throw new ArgumentException("zipPath cannot be null or empty", nameof(zipPath));
+            }
+
+            if (string.IsNullOrEmpty(extractTo))
+            {
+                throw new ArgumentException("extractTo cannot be null or empty", nameof(extractTo));
+            }
+
+            var clientHandler = new HttpClientHandler();
+            clientHandler.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
+            clientHandler.ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => true;
 
-        var fileUrl = @"https://www.ukrposhta.ua/files/shares/out/postindex.zip?_ga=2.154835320.1775361449.1673626930-13
-                           10026737.1673626930&_gl=1*bx4o4d*_ga*MTMxMDAyNjczNy4xNjczNjI2OTMw*_ga_6400KY4HRY*MTY3MzY4MTA5Ny4zLjA
-                           uMTY3MzY4MTA5OC41OS4wLjA.";
+            using (var client = new HttpClient(clientHandler, false) { DefaultRequestHeaders = { }, Timeout = TimeSpan.FromSeconds(60) })
```

Comment:
Please convert block-scoped using to using declaration.

Reply:
done

Reply:
Even though the issue is resolved, I suggest to insert blank lines on every **init** property.